core/src/test/scala/unit/kafka/metrics/MetricsTest.scala

```diff
@@ -121,6 +125,17 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
     // Produce a few messages to create the metrics
     TestUtils.produceMessages(servers, topic, nMessages)
 
+    // Check the log size for each broker so that we can distinguish between failures caused by replication issues
+    // versus failures caused by the metrics
+    val topicPartition = new TopicPartition(topic, 0)
+    servers.foreach { server =>
+      val log = server.logManager.logsByTopicPartition.get(new TopicPartition(topic, 0))
+      val brokerId = server.config.brokerId
+      val logSize = log.map(_.size)
+      assertTrue(s"Expected broker $brokerId to have a Log for $topicPartition with positive size, actual: $logSize",
+        logSize.map(_ > 0).getOrElse(false))
+    }
+
```

Comment:
I deleted my previous code as it was inaccurate. What I meant to say was, we call `removeMetric` for replication bytes in/out even if the metrics are not defined because a topic is defined (@junrao spotted this over the weekend). This should be OK because the tags would be different, but I also changed it to make it more explicit.